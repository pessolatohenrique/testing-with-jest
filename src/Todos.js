import React, { Component } from 'react';
import { connect } from "react-redux";

class Todos extends Component {
    render() {
        return (
            <ul className="Todos">
                {this.props.todos.map(todo => {
                    return <li key={todo.id}>{todo.text}</li>
                })}
            </ul>
        )
    }
}

const mapStateToProps = state => ({
    todos: state.todos
})

export default connect(mapStateToProps)(Todos);