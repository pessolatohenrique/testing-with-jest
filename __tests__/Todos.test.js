import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Todos from '../src/Todos';

const mockStore = configureStore();

const initialState = {
    todos : [
        {id: 0, text: "Trabalhar"},
        {id: 1, text: "Estudar react"},
        {id: 2, text: "Descansar"},
    ]
};

const store = mockStore(initialState);

describe('Testing Todos component', () => {
    it('should render correctyl', () => {
        const wrapper = shallow(
            <Todos />,
            { context: { store }}
        ).dive();

        expect(wrapper.debug()).toMatchSnapshot();
    })
});