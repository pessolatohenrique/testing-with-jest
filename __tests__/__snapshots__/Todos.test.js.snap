// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todos component should render correctyl 1`] = `
"<ul className=\\"Todos\\">
  <li>
    Trabalhar
  </li>
  <li>
    Estudar react
  </li>
  <li>
    Descansar
  </li>
</ul>"
`;
